core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -249,7 +254,7 @@ public long skip(long n) throws IOException {
     }
 
     long toSkip = Math.min(n, remaining());
-    seekInternal(mPos + toSkip);
+    seek(mPos + toSkip);
```

Comment:
Because I think the partially read block caching should be implemented in skip as well.  Previously skip doesn't have this feature. I have not seen a problem related to skip but maybe spark use skip in the future. that will make it very hard to debug if we don't have this